### DISCUSSION TITLE
unpacking archives

### DISCUSSION THREAD

**Discussion by thezoggy** [sabnzbd/sabnzbd#2646] [General]
right now we take a job and break out different file sets...
```
    joinables: List[str] = [],
    zips: List[str] = [],
    rars: List[str] = [],
    sevens: List[str] = [],
```
which are selected by the regex:
```
RAR_RE = re.compile(r"\.(?P<ext>part\d*\.rar|rar|r\d\d|s\d\d|t\d\d|u\d\d|v\d\d|\d\d\d?\d)$", re.I)
SPLITFILE_RE = re.compile(r"\.(\d\d\d?\d$)", re.I)
ZIP_RE = re.compile(r"\.(zip$)", re.I)
SEVENZIP_RE = re.compile(r"\.7z$", re.I)
SEVENMULTI_RE = re.compile(r"\.7z\.\d+$", re.I)
```

then we use unrar for rarsets, unzip for zip, 7zip for 7z

For our purposes, 7zip and unrar both can handle split files / zip / rars / 7z each.

1) Why not actually just have each app (7zip/unrar) able to just do it all in one go instead of trying to segment it out?
  * Get rid of zip
  * Fallback (unrar <-> 7zip) if user disables enable_unrar/unrar/7zip.. 

2) Thoughts on updating the regex for the apps to actually cover the extensions they can cover.. or increase it so (in response to the recent tar question)

unrar can do rar/zip/7z/arj/bz2/cab/gz/lz/lzh/tar/uue/xz/z/zipx/zst/001
Not sure of complete list for 7zip, but i know it can do rar/zip/7z/arj/bzip/cab/gzip/lzh/tar/z/001

**Comment by Safihre** [comment]
We already use 7zip for zip files.

**Reply by thezoggy** [reply]
We try to use 7zip first, then fallback to unzip if 7zip is not enabled.  

So my suggestion above is something similar, get rid of unzip and then just use unrar + 7zip. Use one then fallback to the other if not enabled/has issues.

**Reply by Safihre** [reply]
We don't include unzip in any of the binaries, so it's only used in case it happens to be available on the system and 7zip isn't. Guess we could remove it to just save some code, but that's the only benefit on  that one.
